File: examples/record/prepare_record/PrepareRecord.java
import com.bloock.sdk.client.KeyClient;
import com.bloock.sdk.client.RecordClient;
import com.bloock.sdk.entity.key.ManagedKey;
import com.bloock.sdk.entity.record.Record;

import java.nio.file.Files;
import java.nio.file.Paths;

public class PrepareRecord {
  public static void main(String[] args) throws Exception {
    try {
      RecordClient recordClient = new RecordClient();

      Record record = recordClient.fromString("Hello world").build();

      // we can get the hash of the record
      String hash = record.getHash();
      System.out.println(hash);

      // build a record from an existing record
      Record record2 = recordClient.fromRecord(record).build();

      // we can read a file as an array of bytes
      byte[] file = Files.readAllBytes(Paths.get("sample.pdf"));
      Record record3 = recordClient.fromFile(file).build();

    } catch (Exception e) {
      System.out.println(e);
    }
  }
}

File: examples/record/prepare_record/prepare_record.py
from bloock.client.record import RecordClient

if __name__ == "__main__":
    record_client = RecordClient()
    record = record_client.from_string("Hello world").build()
    # we can get the hash of the record
    record_hash = record.get_hash()

    # build a record from an existing record
    record2 = record_client.from_record(record).build()

    # we can read a file as an array of bytes
    with open("sample.pdf", "rb") as file:
        data = bytes(file.read())
        # and build a record from it
        record3 = record_client.from_file(data).build()

